feat(concrete_core): add a `GlweCiphertextTrivialDecryptionEngine` trait and implementation

aPere3 opened the pull request:
### Resolves:

- zama-ai/concrete_internal#345

### Description

We need the inverse operation of the glwe ciphertext trivial encryption to implement its fixture.

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (apply the `aws_test` to the PR to launch the tests on AWS)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes and add them to commit message following the conventional commit [specification][conventional-breaking]

<!--
### Requires: `<link_your_required_issue_here>`
-->

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

orelcosseron commented on concrete-core/src/backends/core/implementation/engines/glwe_ciphertext_trivial_decryption.rs:
```diff
@@ -0,0 +1,100 @@
+    /// use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+    /// use concrete_core::prelude::*;
+    ///
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
```
Same as above

aPere3 replied:
Done !

aPere3 commented on concrete-core/src/backends/core/implementation/engines/glwe_ciphertext_trivial_decryption.rs:
```diff
@@ -0,0 +1,100 @@
+    /// use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+    /// use concrete_core::prelude::*;
+    ///
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
```
You are right that in this case, it is not super relevant. I wouldn't remove them though, and keep disclaimer in every doctest. This way, there is no way a user can miss a disclaimer when reworking any of the examples, even those which in the first place were not relevant.

aPere3 replied:
Done thanks !

aPere3 commented:
Hello Orel,

Thanks for your review.

This entry point does not feel so mistake prone to me, as we don't provide any secret key in argument. I imagine that decrypting a message without providing a secret key should ring a bell on every serious user of the library.

Lately, with @agnesLeroy, we have been talking about separating the different `*Engine` traits into different sub-modules, to increase the readability of the specification. Maybe we could have a submodule with those unsecure things which emphasize the fact that it should not be used in a secure context ? 

For the fixture, we generally need to have forward and backward functions for every operators providing conversion or encryption. It is the case for this one, but it should be the case for all the others like the fft.

Cheers,

Alex.

agnesLeroy commented:
@aPere3 looks good to me! As suggested by Orel I'd add a small comment in the doc test to emphasize the fact that whatever the parameters chosen the operation is unsecure. And I agree it's a good idea to have this kind of unsafe engines in a dedicated sub module. :smiley:

aPere3 commented:
Hello ! 

I made the changes you requested (I also added a disclaimer in the existing examples of `GlweCiphertextTrivialEncryptionEngine`).

Also, I added the commit for the fixture on top of it.

Best,

Alex
